src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/RuntimeImports.cs

```diff
@@ -877,5 +877,8 @@ internal static IntPtr RhGetModuleSection(TypeManagerHandle module, ReadyToRunSe
         [LibraryImport(RuntimeLibrary)]
         internal static unsafe partial void RhCpuIdEx(int* cpuInfo, int functionId, int subFunctionId);
 #endif
+
+        [LibraryImport(RuntimeLibrary)]
+        internal static unsafe partial void RhSetCurrentThreadName(char* name);
```

Comment:
```suggestion
#if TARGET_UNIX
        [LibraryImport(RuntimeLibrary, StringMarshalling = StringMarshalling.Utf8)]
        internal static unsafe partial void RhSetCurrentThreadName(string name);
#else
        [LibraryImport(RuntimeLibrary, StringMarshalling = StringMarshalling.Utf16)]
        internal static unsafe partial void RhSetCurrentThreadName(string name);
#endif
```
I think we should be able to avoid back and forth string conversion on Windows by marshaling the name differently on Windows vs. non-Windows. (There is on other place in RuntimeImports that does that.)